packages/app/src/systems/Transaction/components/TxRecipientCard/TxRecipientCard.tsx

```diff
@@ -29,8 +40,41 @@ export const TxRecipientCard: TxRecipientCardComponent = ({
   const isContract = recipient?.type === AddressType.contract;
   const isEthChain = recipient?.chain === ChainName.ethereum;
   const isNetwork = address === 'Network';
-  const name =
-    accounts?.find((a) => a.address === fuelAddress)?.name || 'unknown';
+
+  // biome-ignore lint/correctness/useExhaustiveDependencies: <explanation>
+  useEffect(() => {
+    const setAccountName = () => {
+      const accountName =
+        accounts?.find((a) => a.address === fuelAddress)?.name || 'unknown';
+      setName(accountName);
+    };
+
+    const resolveAddressName = async () => {
+      const handleName = await reverseResolver(fuelAddress);
+
+      if (handleName) {
+        return setName(`@${handleName}`);
+      }
+
+      setAccountName();
+    };
+
+    if (!name && isReceiver) {
+      resolveAddressName();
+    }
+
+    if (!isReceiver) {
+      setAccountName();
+    }
+  }, [isReceiver, name]);
+
+  const Name = !name ? (
+    <ContentLoader width={100} height={17} viewBox="0 0 80 17">
+      <ContentLoader.Rect x="0" y="0" width="100" height="17" rx="4" />
+    </ContentLoader>
+  ) : (
+    `${name}`
+  );
```

Comment:
I added a loader to the name to avoid screen flickering when the search is completed.